### PR TITLE
feat: Add organization artifact metadata APIs

### LINKED ISSUE #3917: Support for new Artifact Metadata endpoints
GitHub [announced](https://github.blog/changelog/2026-01-13-new-fine-grained-permission-for-artifact-metadata-is-now-generally-available/) endpoints for artifact metadata

### PR DESCRIPTION
**Pull request by maishivamhoo123**
## Description
This PR implements the Organization Artifact Metadata APIs as requested in #3917. 
This version corrects the scope from Repository to Organization level.

## Changes
- **Logic**: Moved implementation to `github/orgs_artifacts.go`.
- **Endpoints Added**:
    - `POST /orgs/{org}/artifacts/metadata/deployment-record`
    - `POST /orgs/{org}/artifacts/metadata/deployment-record/cluster/{cluster}`
    - `POST /orgs/{org}/artifacts/metadata/storage-record`
    - `GET /orgs/{org}/artifacts/{subject_digest}/metadata/deployment-records`
    - `GET /orgs/{org}/artifacts/{subject_digest}/metadata/storage-records`
- **Testing**: Added comprehensive unit tests in `github/orgs_artifacts_test.go`.
- **Generated**: Updated `github-accessors.go` and `github-stringify_test.go` via `script/generate.sh`.

## Checklist
- [x] I have reviewed my own diff for unnecessary changes.
- [x] All tests pass (`script/test.sh`).
- [x] Code is linted (`script/lint.sh`).
- [x] Ran `script/generate.sh`.

Fixes #3917

### CODE REVIEW

**Comment by maishivamhoo123** [comment]
@gmlewis  Sir can you please review the code?

File: github/orgs_artifacts_test.go [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+
+	r2 := ArtifactDeploymentResponse{TotalCount: Ptr(1)}
+	_ = r2.String()
+
+	r3 := ArtifactStorageRecord{Name: Ptr("n")}
+	_ = r3.String()
+
+	r4 := ArtifactStorageResponse{TotalCount: Ptr(1)}
+	_ = r4.String()
+}
```
Comment (alexandear) [inline github/orgs_artifacts_test.go]: Why do we need this test? It's fully duplicates `TestArtifactDeploymentResponse_String`, `TestArtifactDeploymentRecord_String`, `TestArtifactStorageRecord_String`, `TestArtifactStorageResponse_String`.

File: github/orgs_artifacts.go [hunk 2]
```diff
@@ -0,0 +1,151 @@
+// Copyright 2024 The go-github AUTHORS. All rights reserved.
```
Comment (alexandear) [inline github/orgs_artifacts.go]: Please fix the year here and below.

File: github/orgs_artifacts.go [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+}
+
+func (r ArtifactStorageResponse) String() string { return Stringify(r) }
+
+// CreateArtifactDeploymentRecord creates an artifact deployment record for an organization.
+//
+// GitHub API docs: https://docs.github.com/rest/orgs/artifact-metadata#create-an-artifact-deployment-record
+//
+//meta:operation POST /orgs/{org}/artifacts/metadata/deployment-record
+func (s *OrganizationsService) CreateArtifactDeploymentRecord(ctx context.Context, org string, record *ArtifactDeploymentRecord) (*ArtifactDeploymentResponse, *Response, error) {
```
Comment (Not-Dhananjay-Mishra) [inline github/orgs_artifacts.go]: Will it make sense to make a new struct for request body, In my opinion it will be easy for the users. 
Same apply for `SetClusterDeploymentRecords` and `CreateArtifactStorageRecord`

File: github/orgs_artifacts.go [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+	resp, err := s.client.Do(ctx, req, v)
+	return v, resp, err
+}
+
+// ListArtifactDeploymentRecords lists deployment records for an artifact metadata.
+//
+// GitHub API docs: https://docs.github.com/rest/orgs/artifact-metadata#list-artifact-deployment-records
+//
+//meta:operation GET /orgs/{org}/artifacts/{subject_digest}/metadata/deployment-records
+func (s *OrganizationsService) ListArtifactDeploymentRecords(ctx context.Context, org, subjectDigest string, opts *ListOptions) (*ArtifactDeploymentResponse, *Response, error) {
```
Comment (Not-Dhananjay-Mishra) [inline github/orgs_artifacts.go]:
```suggestion
func (s *OrganizationsService) ListArtifactDeploymentRecords(ctx context.Context, org, subjectDigest string) (*ArtifactDeploymentResponse, *Response, error) {
```
According to docs this endpoint doesn't support pagination

Image

File: github/orgs_artifacts.go [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+	resp, err := s.client.Do(ctx, req, v)
+	return v, resp, err
+}
+
+// ListArtifactStorageRecords lists artifact storage records with a given subject digest.
+//
+// GitHub API docs: https://docs.github.com/rest/orgs/artifact-metadata#list-artifact-storage-records
+//
+//meta:operation GET /orgs/{org}/artifacts/{subject_digest}/metadata/storage-records
+func (s *OrganizationsService) ListArtifactStorageRecords(ctx context.Context, org, subjectDigest string, opts *ListOptions) (*ArtifactStorageResponse, *Response, error) {
```
Comment (Not-Dhananjay-Mishra) [inline github/orgs_artifacts.go]:
```suggestion
func (s *OrganizationsService) ListArtifactStorageRecords(ctx context.Context, org, subjectDigest string) (*ArtifactStorageResponse, *Response, error) {
```
According to docs this endpoint doesn't support pagination

Image

**Review by gmlewis — APPROVED** [review]
Thank you, @maishivamhoo123, @alexandear , and @Not-Dhananjay-Mishra!
LGTM.
Awaiting second LGTM+Approval from any other contributor to this repo before merging.

cc: @stevehipwell - @zyfy29

File: github/orgs_artifacts.go [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+}
+
+func (r ArtifactStorageResponse) String() string { return Stringify(r) }
+
+// CreateArtifactDeploymentRecord creates an artifact deployment record for an organization.
+//
+// GitHub API docs: https://docs.github.com/rest/orgs/artifact-metadata#create-an-artifact-deployment-record
+//
+//meta:operation POST /orgs/{org}/artifacts/metadata/deployment-record
+func (s *OrganizationsService) CreateArtifactDeploymentRecord(ctx context.Context, org string, record *ArtifactDeploymentRecord) (*ArtifactDeploymentResponse, *Response, error) {
```
Comment (Not-Dhananjay-Mishra) [inline github/orgs_artifacts.go]: This endpoint request body have some missing items in current implantation - `version`, `runtime_risks` and `github_repository`

Should we define a new struct to properly represent the complete request body?

Image

File: github/orgs_artifacts.go [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+	resp, err := s.client.Do(ctx, req, v)
+	return v, resp, err
+}
+
+// SetClusterDeploymentRecords sets deployment records for a given cluster.
+//
+// GitHub API docs: https://docs.github.com/rest/orgs/artifact-metadata#set-cluster-deployment-records
+//
+//meta:operation POST /orgs/{org}/artifacts/metadata/deployment-record/cluster/{cluster}
+func (s *OrganizationsService) SetClusterDeploymentRecords(ctx context.Context, org, cluster string, record *ArtifactDeploymentRecord) (*ArtifactDeploymentResponse, *Response, error) {
```
Comment (Not-Dhananjay-Mishra) [inline github/orgs_artifacts.go]: This endpoint request body have some missing item in current implantation - `deployments` (array of object)

Should we define a new struct to properly represent the complete request body?

Image

File: github/orgs_artifacts.go [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+	resp, err := s.client.Do(ctx, req, v)
+	return v, resp, err
+}
+
+// CreateArtifactStorageRecord creates metadata storage records for artifacts.
+//
+// GitHub API docs: https://docs.github.com/rest/orgs/artifact-metadata#create-artifact-metadata-storage-record
+//
+//meta:operation POST /orgs/{org}/artifacts/metadata/storage-record
+func (s *OrganizationsService) CreateArtifactStorageRecord(ctx context.Context, org string, record *ArtifactStorageRecord) (*ArtifactStorageResponse, *Response, error) {
```
Comment (Not-Dhananjay-Mishra) [inline github/orgs_artifacts.go]: This endpoint request body have some missing item in current implantation - `version`, `path` and `github_repository`

Image

File: github/orgs_artifacts.go (hunk 7 above)
Comment (maishivamhoo123) [reply github/orgs_artifacts.go]: @Not-Dhananjay-Mishra yaa i was thinking to do this in previous pr but I missed it . this is looking good to me i will do this is in . 
Thank you .

**Comment by maishivamhoo123** [comment]
@Not-Dhananjay-Mishra can you please check once i think the latest pr is able to meet all your requirements?

File: github/orgs_artifacts.go [hunk 9]
```diff
@@ -0,0 +1,159 @@
...
+}
+
+func (r ArtifactStorageResponse) String() string { return Stringify(r) }
+
+// CreateArtifactDeploymentRecord creates an artifact deployment record for an organization.
+//
+// GitHub API docs: https://docs.github.com/rest/orgs/artifact-metadata#create-an-artifact-deployment-record
+//
+//meta:operation POST /orgs/{org}/artifacts/metadata/deployment-record
+func (s *OrganizationsService) CreateArtifactDeploymentRecord(ctx context.Context, org string, record *ArtifactDeploymentRecord) (*ArtifactDeploymentResponse, *Response, error) {
```
Comment (alexandear) [inline github/orgs_artifacts.go]: @maishivamhoo123 this comment is not addressed: https://github.com/google/go-github/pull/3944/changes#r2732556939

Should be a separate struct for the request body.

**Review by gmlewis — APPROVED** [review]
Thank you, @maishivamhoo123!
LGTM.
Awaiting second LGTM+Approval from any other contributor to this repo before merging.

cc: @stevehipwell - @alexandear - @zyfy29 - @Not-Dhananjay-Mishra

File: github/orgs_artifacts.go [hunk 10]
```diff
@@ -0,0 +1,200 @@
...
+	TotalCount        *int                        `json:"total_count,omitempty"`
+	DeploymentRecords []*ArtifactDeploymentRecord `json:"deployment_records,omitempty"`
+}
+
+// ClusterDeploymentRecordsRequest represents the request body for setting cluster deployment records.
+type ClusterDeploymentRecordsRequest struct {
+	LogicalEnvironment  *string                            `json:"logical_environment,omitempty"`
+	PhysicalEnvironment *string                            `json:"physical_environment,omitempty"`
+	Deployments         []*CreateArtifactDeploymentRequest `json:"deployments,omitempty"`
+}
```
Comment (Not-Dhananjay-Mishra) [inline github/orgs_artifacts.go:76]: These parameters are required for [Set cluster deployment records](https://docs.github.com/en/rest/orgs/artifact-metadata?apiVersion=2022-11-28#set-cluster-deployment-records):
```suggestion
// ClusterDeploymentRecordsRequest represents the request body for setting cluster deployment records.
type ClusterDeploymentRecordsRequest struct {
	LogicalEnvironment  string                            `json:"logical_environment"`
	PhysicalEnvironment *string                            `json:"physical_environment,omitempty"`
	Deployments         []*CreateArtifactDeploymentRequest `json:"deployments"`
}
```

File: github/orgs_artifacts.go [hunk 11]
```diff
@@ -0,0 +1,200 @@
...
+	Name             *string `json:"name,omitempty"`
+	Digest           *string `json:"digest,omitempty"`
+	Version          *string `json:"version,omitempty"`
+	ArtifactURL      *string `json:"artifact_url,omitempty"`
+	Path             *string `json:"path,omitempty"`
+	RegistryURL      *string `json:"registry_url,omitempty"`
+	Repository       *string `json:"repository,omitempty"`
+	Status           *string `json:"status,omitempty"`
+	GithubRepository *string `json:"github_repository,omitempty"`
+}
```
Comment (Not-Dhananjay-Mishra) [inline github/orgs_artifacts.go:102]: These parameters are required for [Create artifact metadata storage record](https://docs.github.com/en/rest/orgs/artifact-metadata?apiVersion=2022-11-28#create-artifact-metadata-storage-record):
```suggestion
type CreateArtifactStorageRequest struct {
	Name             string `json:"name"`
	Digest           string `json:"digest"`
	Version          *string `json:"version,omitempty"`
	ArtifactURL      *string `json:"artifact_url,omitempty"`
	Path             *string `json:"path,omitempty"`
	RegistryURL      string `json:"registry_url"`
	Repository       *string `json:"repository,omitempty"`
	Status           *string `json:"status,omitempty"`
	GithubRepository *string `json:"github_repository,omitempty"`
}
```

File: github/orgs_artifacts.go [hunk 12]
```diff
@@ -0,0 +1,200 @@
...
+	Version             *string           `json:"version,omitempty"`
+	Status              *string           `json:"status,omitempty"`
+	LogicalEnvironment  *string           `json:"logical_environment,omitempty"`
+	PhysicalEnvironment *string           `json:"physical_environment,omitempty"`
+	Cluster             *string           `json:"cluster,omitempty"`
+	DeploymentName      *string           `json:"deployment_name,omitempty"`
+	Tags                map[string]string `json:"tags,omitempty"`
+	RuntimeRisks        []string          `json:"runtime_risks,omitempty"`
+	GithubRepository    *string           `json:"github_repository,omitempty"`
+}
```
Comment (Not-Dhananjay-Mishra) [inline github/orgs_artifacts.go:51]: These parameters are required for both [Create an artifact deployment record](https://docs.github.com/en/rest/orgs/artifact-metadata?apiVersion=2022-11-28#create-an-artifact-deployment-record) and [Set cluster deployment records](https://docs.github.com/en/rest/orgs/artifact-metadata?apiVersion=2022-11-28#set-cluster-deployment-records)
```suggestion
type CreateArtifactDeploymentRequest struct {
	Digest              string           `json:"digest"`
	Name                string           `json:"name"`
	Version             *string           `json:"version,omitempty"`
	Status              *string           `json:"status,omitempty"`
	LogicalEnvironment  *string           `json:"logical_environment,omitempty"`
	PhysicalEnvironment *string           `json:"physical_environment,omitempty"`
	Cluster             *string           `json:"cluster,omitempty"`
	DeploymentName      string           `json:"deployment_name"`
	Tags                map[string]string `json:"tags,omitempty"`
	RuntimeRisks        []string          `json:"runtime_risks,omitempty"`
	GithubRepository    *string           `json:"github_repository,omitempty"`
}
```

File: github/orgs_artifacts.go [hunk 13]
```diff
@@ -0,0 +1,200 @@
...
+	}
+	return v, resp, nil
+}
+
+// ListArtifactDeploymentRecords lists deployment records for an artifact metadata.
+//
+// GitHub API docs: https://docs.github.com/rest/orgs/artifact-metadata#list-artifact-deployment-records
+//
+//meta:operation GET /orgs/{org}/artifacts/{subject_digest}/metadata/deployment-records
+func (s *OrganizationsService) ListArtifactDeploymentRecords(ctx context.Context, org, subjectDigest string, opts *ListOptions) (*ArtifactDeploymentResponse, *Response, error) {
```
Comment (Not-Dhananjay-Mishra) [inline github/orgs_artifacts.go]: According to docs this endpoint doesn't support pagination [comment](https://github.com/google/go-github/pull/3944#discussion_r2732569788)
```suggestion
func (s *OrganizationsService) ListArtifactDeploymentRecords(ctx context.Context, org, subjectDigest string) (*ArtifactDeploymentResponse, *Response, error) {
```

File: github/orgs_artifacts.go [hunk 14]
```diff
@@ -0,0 +1,200 @@
...
+	}
+	return v, resp, nil
+}
+
+// ListArtifactStorageRecords lists artifact storage records with a given subject digest.
+//
+// GitHub API docs: https://docs.github.com/rest/orgs/artifact-metadata#list-artifact-storage-records
+//
+//meta:operation GET /orgs/{org}/artifacts/{subject_digest}/metadata/storage-records
+func (s *OrganizationsService) ListArtifactStorageRecords(ctx context.Context, org, subjectDigest string, opts *ListOptions) (*ArtifactStorageResponse, *Response, error) {
```
Comment (Not-Dhananjay-Mishra) [inline github/orgs_artifacts.go]: According to docs this endpoint doesn't support pagination [comment](https://github.com/google/go-github/pull/3944#discussion_r2732575783)
```suggestion
func (s *OrganizationsService) ListArtifactStorageRecords(ctx context.Context, org, subjectDigest string) (*ArtifactStorageResponse, *Response, error) {
```

File: github/orgs_artifacts.go (hunk 13 above)
Comment (alexandear) [reply github/orgs_artifacts.go]: Thank you for the detailed review.

I think Copilot can catch such obvious mistakes. @gmlewis, could we set up Copilot or any other agent-based review tool for this repo?

Comment (gmlewis) [reply github/orgs_artifacts.go]: @alexandear - I don't have full admin privileges to this repo, but I can ask.

@google-admin - can Copilot or any other agent-based automated code review tool be added to this repo?

File: github/orgs_artifacts.go [hunk 15]
```diff
@@ -0,0 +1,192 @@
...
+	PhysicalEnvironment *string           `json:"physical_environment,omitempty"`
+	Cluster             *string           `json:"cluster,omitempty"`
+	DeploymentName      *string           `json:"deployment_name,omitempty"`
+	Tags                map[string]string `json:"tags,omitempty"`
+	RuntimeRisks        []string          `json:"runtime_risks,omitempty"`
+	GithubRepository    *string           `json:"github_repository,omitempty"`
+	AttestationID       *int64            `json:"attestation_id,omitempty"`
+	CreatedAt           *Timestamp        `json:"created_at,omitempty"`
+	UpdatedAt           *Timestamp        `json:"updated_at,omitempty"`
+}
```
Comment (Not-Dhananjay-Mishra) [inline github/orgs_artifacts.go:36]: How about removing the items which are not returned by endpoint as you have already made a new struct for request body.

```suggestion
type ArtifactDeploymentRecord struct {
	ID                  *int64            `json:"id,omitempty"`
	Digest              *string           `json:"digest,omitempty"`
	LogicalEnvironment  *string           `json:"logical_environment,omitempty"`
	PhysicalEnvironment *string           `json:"physical_environment,omitempty"`
	Cluster             *string           `json:"cluster,omitempty"`
	DeploymentName      *string           `json:"deployment_name,omitempty"`
	Tags                map[string]string `json:"tags,omitempty"`
	RuntimeRisks        []string          `json:"runtime_risks,omitempty"`
	AttestationID       *int64            `json:"attestation_id,omitempty"`
	CreatedAt           *Timestamp        `json:"created_at,omitempty"`
	UpdatedAt           *Timestamp        `json:"updated_at,omitempty"`
}
```

File: github/orgs_artifacts.go [hunk 16]
```diff
@@ -0,0 +1,192 @@
...
+	Version          *string    `json:"version,omitempty"`
+	ArtifactURL      *string    `json:"artifact_url,omitempty"`
+	Path             *string    `json:"path,omitempty"`
+	RegistryURL      *string    `json:"registry_url,omitempty"`
+	Repository       *string    `json:"repository,omitempty"`
+	Status           *string    `json:"status,omitempty"`
+	GithubRepository *string    `json:"github_repository,omitempty"`
+	CreatedAt        *Timestamp `json:"created_at,omitempty"`
+	UpdatedAt        *Timestamp `json:"updated_at,omitempty"`
+}
```
Comment (Not-Dhananjay-Mishra) [inline github/orgs_artifacts.go:89]: same...
```suggestion
type ArtifactStorageRecord struct {
	ID               *int64     `json:"id,omitempty"`
	Name             *string    `json:"name,omitempty"`
	Digest           *string    `json:"digest,omitempty"`
	ArtifactURL      *string    `json:"artifact_url,omitempty"`
	RegistryURL      *string    `json:"registry_url,omitempty"`
	Repository       *string    `json:"repository,omitempty"`
	Status           *string    `json:"status,omitempty"`
	CreatedAt        *Timestamp `json:"created_at,omitempty"`
	UpdatedAt        *Timestamp `json:"updated_at,omitempty"`
}
```

**Comment by maishivamhoo123** [comment]
@alexandear , @Not-Dhananjay-Mishra   , @gmlewis can you please review the changes , I made all the suggested changes .

File: github/orgs_artifacts.go [hunk 17]
```diff
@@ -0,0 +1,185 @@
...
+	TotalCount     *int                     `json:"total_count,omitempty"`
+	StorageRecords []*ArtifactStorageRecord `json:"storage_records,omitempty"`
+}
+
+// CreateArtifactDeploymentRecord creates an artifact deployment record for an organization.
+//
+// GitHub API docs: https://docs.github.com/rest/orgs/artifact-metadata#create-an-artifact-deployment-record
+//
+//meta:operation POST /orgs/{org}/artifacts/metadata/deployment-record
+func (s *OrganizationsService) CreateArtifactDeploymentRecord(ctx context.Context, org string, record *CreateArtifactDeploymentRequest) (*ArtifactDeploymentResponse, *Response, error) {
```
Comment (alexandear) [inline github/orgs_artifacts.go]: `record` is required. Can we pass it by value?


```suggestion
func (s *OrganizationsService) CreateArtifactDeploymentRecord(ctx context.Context, org string, record CreateArtifactDeploymentRequest) (*ArtifactDeploymentResponse, *Response, error) {
```

Comment (alexandear) [reply github/orgs_artifacts.go]: The same for other endpoints

**Comment by gmlewis** [comment]
@maishivamhoo123 - it looks to me like you addressed all the review feedback... (thank you!), but just to be sure, let's let @alexandear respond, and then we will merge.

File: github/orgs_artifacts.go [hunk 18]
```diff
@@ -0,0 +1,185 @@
...
+	GithubRepository *string `json:"github_repository,omitempty"`
+}
+
+// ArtifactStorageResponse represents the response for storage records.
+type ArtifactStorageResponse struct {
+	TotalCount     *int                     `json:"total_count,omitempty"`
+	StorageRecords []*ArtifactStorageRecord `json:"storage_records,omitempty"`
+}
+
+// CreateArtifactDeploymentRecord creates an artifact deployment record for an organization.
```
Comment (alexandear) [inline github/orgs_artifacts.go]: From the doc:

Image

```suggestion
// CreateArtifactDeploymentRecord creates or updates deployment records for an artifact associated with an organization.
```

File: github/orgs_artifacts.go [hunk 19]
```diff
@@ -0,0 +1,185 @@
...
+	Name                string            `json:"name"`
+	Version             *string           `json:"version,omitempty"`
+	Status              *string           `json:"status,omitempty"`
+	LogicalEnvironment  *string           `json:"logical_environment,omitempty"`
+	PhysicalEnvironment *string           `json:"physical_environment,omitempty"`
+	Cluster             *string           `json:"cluster,omitempty"`
+	DeploymentName      string            `json:"deployment_name"`
+	Tags                map[string]string `json:"tags,omitempty"`
+	RuntimeRisks        []string          `json:"runtime_risks,omitempty"`
+	GithubRepository    *string           `json:"github_repository,omitempty"`
```
Comment (alexandear) [inline github/orgs_artifacts.go]: 1. Could we sort fields as they appear in the docs?

```
name
digest
version
status
logical_environment
physical_environment
cluster
deployment_name
tags
runtime_risks
github_repository
```

2. `status`, `logical_environment` are required params and should be without `omitempty`.

File: github/orgs_artifacts.go [hunk 20]
```diff
@@ -0,0 +1,185 @@
...
+type CreateArtifactDeploymentRequest struct {
+	Digest              string            `json:"digest"`
+	Name                string            `json:"name"`
+	Version             *string           `json:"version,omitempty"`
+	Status              *string           `json:"status,omitempty"`
+	LogicalEnvironment  *string           `json:"logical_environment,omitempty"`
+	PhysicalEnvironment *string           `json:"physical_environment,omitempty"`
+	Cluster             *string           `json:"cluster,omitempty"`
+	DeploymentName      string            `json:"deployment_name"`
+	Tags                map[string]string `json:"tags,omitempty"`
```
Comment (alexandear) [inline github/orgs_artifacts.go]: `Tags` has `Object` type. Why is it represented as `map[string]string`? Maybe it should be `map[string]any`?

```suggestion
	Tags                map[string]any `json:"tags,omitempty"`
```

For example, we have `Inputs` as `Object` [here](https://docs.github.com/en/rest/actions/workflows?apiVersion=2022-11-28#create-a-workflow-dispatch-event):
https://github.com/google/go-github/blob/5d70283e9d97fae8befdfc16e19f02992e909e67/github/actions_workflows.go#L56

File: github/orgs_artifacts.go [hunk 21]
```diff
@@ -0,0 +1,185 @@
...
+// ArtifactDeploymentRecord represents a GitHub artifact deployment record.
+type ArtifactDeploymentRecord struct {
+	ID                  *int64            `json:"id,omitempty"`
+	Digest              *string           `json:"digest,omitempty"`
+	LogicalEnvironment  *string           `json:"logical_environment,omitempty"`
+	PhysicalEnvironment *string           `json:"physical_environment,omitempty"`
+	Cluster             *string           `json:"cluster,omitempty"`
+	DeploymentName      *string           `json:"deployment_name,omitempty"`
+	Tags                map[string]string `json:"tags,omitempty"`
+	RuntimeRisks        []string          `json:"runtime_risks,omitempty"`
```
Comment (alexandear) [inline github/orgs_artifacts.go]: Should we create the enum `DeploymentRuntimeRisk` and use it instead of `string`?

```
          "runtime_risks": {
            "type": "array",
            "description": "A list of runtime risks associated with the deployment.",
            "maxItems": 4,
            "uniqueItems": true,
            "items": {
              "type": "string",
              "enum": [
                "critical-resource",
                "internet-exposed",
                "lateral-movement",
                "sensitive-data"
              ]
            }
```

File: github/orgs_artifacts.go [hunk 22]
```diff
@@ -0,0 +1,185 @@
...
+type ArtifactDeploymentResponse struct {
+	TotalCount        *int                        `json:"total_count,omitempty"`
+	DeploymentRecords []*ArtifactDeploymentRecord `json:"deployment_records,omitempty"`
+}
+
+// ClusterDeploymentRecordsRequest represents the request body for setting cluster deployment records.
+type ClusterDeploymentRecordsRequest struct {
+	LogicalEnvironment  string                             `json:"logical_environment"`
+	PhysicalEnvironment *string                            `json:"physical_environment,omitempty"`
+	Deployments         []*CreateArtifactDeploymentRequest `json:"deployments"`
```
Comment (alexandear) [inline github/orgs_artifacts.go]: We can't use `CreateArtifactDeploymentRequest` here. This struct differs from what is expected according to the docs:

Image

File: github/orgs_artifacts.go [hunk 23]
```diff
@@ -0,0 +1,185 @@
...
+	}
+	return v, resp, nil
+}
+
+// ListArtifactDeploymentRecords lists deployment records for an artifact metadata.
+//
+// GitHub API docs: https://docs.github.com/rest/orgs/artifact-metadata#list-artifact-deployment-records
+//
+//meta:operation GET /orgs/{org}/artifacts/{subject_digest}/metadata/deployment-records
+func (s *OrganizationsService) ListArtifactDeploymentRecords(ctx context.Context, org, subjectDigest string) (*ArtifactDeploymentResponse, *Response, error) {
```
Comment (alexandear) [inline github/orgs_artifacts.go:174]: I propose to extend the comment with info: `subjectDigest is SHA256 digest of the artifact, in the form sha256:HEX_DIGEST.`

Comment (alexandear) [reply github/orgs_artifacts.go:174]: the same for `ListArtifactStorageRecords`

**Comment by maishivamhoo123** [comment]
@alexandear Sir Thank you Sir for your review I made all the suggested changes can you please check ?
Thank you sir!

**Comment by gmlewis** [comment]
Thank you, @alexandear!
Merging.

**Comment by maishivamhoo123** [comment]
@alexandear, @gmlewis, @Not-Dhananjay-Mishra Thank you very much for all the reviews and feedback. I truly learned a lot while working on this issue. I’m looking forward to contributing more and learning from your experience. Thanks again!
